lib/ansible/modules/cloud/azure/azure_rm_image_facts.py

```diff
@@ -0,0 +1,268 @@
+        "resource_group": "test-resource-group",
+        "source": "/subscriptions/xyz/resourceGroups/test-resource-group/providers/Microsoft.Compute/virtualMachines/testimagefacts",
+        "tags": null
+    }]
+
+'''
+
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+except:
+    # This is handled in azure_rm_common
+    pass
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+
+AZURE_ENUM_MODULES = ['azure.mgmt.compute.models']
+
+
+class AzureRMImageFacts(AzureRMModuleBase):
+
+    def __init__(self, **kwargs):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str'),
+            name=dict(type='str'),
+            tags=dict(type='list'),
+            format=dict(
+                type='str',
+                choices=['curated',
+                         'raw'],
+                default='raw'
+            )
+
+        )
+
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_images=[])
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.format = None
+        self.tags = None
+
+        super(AzureRMImageFacts, self).__init__(
+            derived_arg_spec=self.module_arg_spec,
+            supports_tags=False,
+            facts_module=True
+        )
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        if self.name and self.resource_group:
+            self.results['ansible_facts']['azure_images'] = self.get_image(self.resource_group, self.name)
+        elif self.name and not self.resource_group:
+            self.results['ansible_facts']['azure_images'] = self.list_images(self.name)
+        elif not self.name and self.resource_group:
+            self.results['ansible_facts']['azure_images'] = self.list_images_by_resource_group(self.resource_group)
+        elif not self.name and not self.resource_group:
+            self.results['ansible_facts']['azure_images'] = self.list_images()
+
+        if self.format == 'curated':
+            for index in range(len(self.results['ansible_facts']['azure_images'])):
+                image = self.results['ansible_facts']['azure_images'][index]
+
+                data_disks = image['properties']['storageProfile']['dataDisks']
+                for disk_index in range(len(data_disks)):
+                    old_disk = data_disks[disk_index]
+                    new_disk = {
+                        'lun': old_disk['lun'],
+                        'disk_size_gb': old_disk['diskSizeGB'],
+                        'managed_disk': old_disk['managedDisk']['id'],
+                        'caching': old_disk['caching'],
+                        'storage_account_type': old_disk['storageAccountType']
+                    }
+                    data_disks[disk_index] = new_disk
+
+                updated = {
+                    'resource_group': self.resource_group,
```

Comment:
resource group should be parsed from image.id, considering the list_all scenario.